instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ManifestResourceProvider.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.resources;
+
+import static java.util.logging.Level.WARNING;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.semconv.ResourceAttributes;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.jar.Manifest;
+import java.util.logging.Logger;
+
+/**
+ * A {@link ResourceProvider} that will attempt to detect the <code>service.name</code> and <code>
+ * service.version</code> from META-INF/MANIFEST.MF.
+ */
+@AutoService(ResourceProvider.class)
+public final class ManifestResourceProvider extends AttributeResourceProvider<Manifest> {
+
+  private static final Logger logger = Logger.getLogger(ManifestResourceProvider.class.getName());
+
+  public ManifestResourceProvider() {
+    this(new SystemHelper());
+  }
+
+  // Visible for testing
+  ManifestResourceProvider(SystemHelper systemHelper) {
+    super(
+        new AttributeProvider<Manifest>() {
+          @Override
+          public Optional<Manifest> readData() {
+            return Optional.ofNullable(
+                    systemHelper.openClasspathResource("META-INF", "MANIFEST.MF"))
```

Comment:
which `MANIFEST.MF` will this get when dependencies are directly on the class path?